packages/contracts/contracts/test/SupplyVaultsTest.t.sol

```diff
@@ -123,10 +126,67 @@ contract SupplyVaultsTest is BaseTest {
     vm.stopPrank();
   }
 
+  function upgradeIflr() internal {
+    IonicFlywheelLensRouter upgradedIflr = new IonicFlywheelLensRouter(PoolDirectory(ap.getAddress("PoolDirectory")));
+    vm.prank(ap.owner());
+    ap.setAddress("IonicFlywheelLensRouter", address(upgradedIflr));
+
+    IonicFlywheel newFwImpl = new IonicFlywheel();
+
+    ProxyAdmin proxyAdmin;
+
+    // replace all flywheels
+    PoolDirectory.Pool[] memory pools = upgradedIflr.fpd().getAllPools();
+    for (uint8 i = 0; i < pools.length; i++) {
+      IonicComptroller pool = IonicComptroller(pools[i].comptroller);
+      address[] memory flywheels = pool.getAccruingFlywheels();
+      for (uint8 j = 0; j < flywheels.length; j++) {
+        // TODO figure out why
+        if (flywheels[j] == 0x2DC3f7B18e8F62F7fE7819596D15E521EEf3b1ec) {
+          proxyAdmin = ProxyAdmin(0xd122669FeF7e62Aa5Df85e945b68dd0B02A42343);
+        }
+        else if (flywheels[j] == 0xcC11Fc7048db155F691Cc20Ac9958Fc465fa0062) {
+          proxyAdmin = ProxyAdmin(0x4De2d8ef97D19def01f236b7a12e5Fb39c087b56);
+        }
+        else if (flywheels[j] == 0x6AfCca37CC93DB6bed729d20ADF203290d465df5 || flywheels[j] == 0x4E854cde138495a3eB9CFe48e50F12dC352cD834) {
+          proxyAdmin = ProxyAdmin(0xaF9cc7599DEFd86226e0f3A6810c4976E4a10f83);
```

Comment:
for some reason the above flywheels were upgraded with different ProxyAdmin contracts